### PR TITLE
Create default.nix

### PR DESCRIPTION
**Pull request by Rakesh4G**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/tiledb/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{  pkgs ? import ./pkgs.nix  }:
+with pkgs;
+let
+  python = python36;
+ 
+  tiledb  = import ./tiledb.nix { inherit pkgs; };
```
Comment (CMCDragonkai) [inline pkgs/development/python-modules/tiledb/default.nix]: You cannot do this for nixpkgs mainline. You need to get the package dependencies as a function.

See how other nixpkgs python modules are structured. Look at `numpy/default.nix`.

File: pkgs/development/python-modules/tiledb/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+in
+  python.pkgs.buildPythonPackage rec {
+    inherit python;
+
+    pname = "tiledb";
+    version = "0.4.2";
+
+    src = pkgs.fetchFromGitHub {
```
Comment (CMCDragonkai) [inline pkgs/development/python-modules/tiledb/default.nix]: Usually nixpkgs mainline prefers you fetch it from PyPi. See how this is done here: https://github.com/NixOS/nixpkgs/pull/54404/files

**Comment by CMCDragonkai** [comment]
The name of the PR and commit should follow the same structure: https://github.com/NixOS/nixpkgs/pull/54404

**Comment by CMCDragonkai** [comment]
Note that this PR relies on the existence of `tiledb` library.

File: pkgs/development/python-modules/tiledb/default.nix (hunk 1 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/tiledb/default.nix]: ok

File: pkgs/development/python-modules/tiledb/default.nix (hunk 2 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/tiledb/default.nix]: ok

**Comment by Rakesh4G** [comment]
Hi Roger, I had closed this request. As i was trying to squash the multiple commits into single commit. But accidentally it is now showing all more than 3500 changes. I will create a fresh PR for tiledb pythonPackage.

**Pull request closed without merging**